### ISSUE TITLE
Wrong values generated by Duration#toISOString() when mixing positive and negative values

### ISSUE THREAD

**Issue by adobo** [moment/moment#2154]
Creating a moment Duration with more than one unit and specifying positive and negative values makes `toISOString()` to return wrong values.

For instance ([jsfiddle](http://jsfiddle.net/k4jwmac4/)):

``` javascript

moment.duration({
  days: -1,
  hours: 8
}).toISOString();
```

Will return `-P1DT8H`, which is not what the duration object contained. `-P1DT8H` would mean _1 day and 8 hours less_, opposed to the original meaning _1 day less, 8 hours more_.

I found this issue while working with FullCalendar, which returns moment Durations on some callbacks. Just a question to know if this should be reported to FullCalendar instead: does it make sense to create Duration objects mixing units and positive/negative values?

Putting it another way: should the unit recalculations be performed before calling the Duration constructor, o should Duration do some calculations before returning an ISO string?

**Comment by coltrane** [comment]
This same issue occurs when using the add/subtract functions to do math on duration objects:

```
> d = moment.duration(3, 'seconds');
Duration {_milliseconds: 3000, _days: 0, _months: 0, _data: Object, _locale: Locale}
> d.subtract(1, 'days');
Duration {_milliseconds: 3000, _days: -1, _months: 0, _data: Object, _locale: Locale}
> d.toISOString();
"-P1DT3S"
> d.toString();
"-P1DT3S"
```

Strangely, the internal attributes of `_data` are correct after the math operation.  And `asSeconds` returns the correct value, even as `toISOString()` continues to report the wrong ISO duration.
```
> d._data;
Object {milliseconds: -0, seconds: -57, minutes: -59, hours: -23, days: 0…}
> d.asSeconds();
-86397
> d.toISOString();
"-P1DT3S"
```

**Comment by ashsearle** [comment]
Should be fixed by pull request #4062, where:

```js
moment.duration({
  days: -1,
  hours: 8
}).toISOString();
```

will return `-P1DT-8H`

**Comment by marwahaha** [comment]
Fixed!
